src/services/group.service.ts

```diff
@@ -0,0 +1,68 @@
+import { singleton } from 'tsyringe';
+
+import { sequelize } from '../data-access/database';
+import { IGroup } from '../models/group.interface';
+import { GroupModel } from '../models/group.model';
+import { UserGroupModel } from '../models/user.model';
+
+@singleton()
+export class GroupService {
+  public async getGroupById(id: string): Promise<IGroup | null> {
+    const group: IGroup | null = await GroupModel.findByPk(id);
+
+    return group;
+  }
+
+  public async getAllGroups(): Promise<IGroup[]> {
+    const groups: IGroup[] = await GroupModel.findAll();
+
+    return groups;
+  }
+
+  public async updateGroup(id: string, groupUpdates: IGroup): Promise<IGroup | null | undefined> {
+    const [numberOfItems, updatedItems]: [number, IGroup[]] = await GroupModel.update(groupUpdates, {
+      where: {
+        id,
+      },
+    });
+
+    return numberOfItems ? updatedItems.shift() : null;
```

Comment:
Looks as you can use destruction for getting first element

Reply:
refactored